pilot/pkg/model/push_context.go

```diff
@@ -419,6 +386,38 @@ func (ps *PushContext) VirtualServices(proxy *Proxy, gateways map[string]bool) [
 	return out
 }
 
+// GetSidecarScope returns a sidecar rule applicable to the config
+// namespace associated with the proxy
+func (ps *PushContext) GetSidecarScope(proxy *Proxy, proxyInstances []*ServiceInstance) *SidecarScope {
+
+	var workloadLabels LabelsCollection
+	for _, w := range proxyInstances {
+		workloadLabels = append(workloadLabels, w.Labels)
+	}
+
+	if sidecars, ok := ps.sidecarsByNamespace[proxy.ConfigNamespace]; ok {
```

Comment:
One question： proxies deployed long ago without ConfigNamespace, the Sidecar will not take effect. 
Better document it explicitly.

Reply:
Well - that's because the code to extract the namespace from nodeid was not added. 
There is no reason to have this limitation.

Reply:
They will all default to the default namespace which wont have a sidecar object. So we generate a default sidecar object which has same behavior as before

Reply:
Why wouldn't default have a sidecar ? And that's a guaranteed way to generate bad config - the proxy is in namespace "foo", which has Sidecar - we can't generate a config for default namespace. 

Not sure what is the reason to ignore the namespace we get in the nodeid - and use the wrong namespace.

Reply:
if you can solve it in a platform neutral way, lets go for it. as it stands now, taking the second piece of a DNS domain name is not guaranteed to be compatible across platforms like cloud foundry where DNS names have no relation to the namespace of the proxy

Reply:
Anyway, I have added code to parse the namespace from the proxy's DNSDomain for older 1.0.x proxies. So for older proxy deployments, we will still get the proxy's namespace and hence we should be able to pick the Sidecar config for the namespace.